app.js

```diff
@@ -158,45 +149,36 @@ define(function(require) {
 
 		bindEvents: function(parent) {
 			var self = this,
-				container = parent.find('.right-content'),
-				bindDefaultNavigation = function defaultNavigation() {
-					parent.find('.left-menu').on('click', '.category:not(.loading)', function() {
-						// Get the ID of the submodule to render
-						var $this = $(this),
-							args = {
-								parent: container,
-								callback: function() {
-									parent.find('.category').removeClass('loading');
-								}
-							},
-							id = $this.attr('id');
-
-						// Display the category we clicked as active
-						parent
-							.find('.category')
-							.removeClass('active')
-							.addClass('loading');
-						$this.toggleClass('active');
-
-						// Empty the main container and then render the submodule content
-						container.empty();
-						monster.pub('voip.' + id + '.render', args);
-					});
-				},
-				bindEventNavigation = function bindEventNavigation() {
-					window.addEventListener('message', function(event) {
-						var args = { parent: container };
-
-						if (event.data.type === 'smartpbx.routing' && self.appFlags.common.navigationMenus.includes(event.data.data)) {
-							container.empty();
-							monster.pub('voip.' + event.data.data + '.render', args);
+				container = parent.find('.right-content');
+
+			parent.find('.left-menu').on('click', '.category:not(.loading)', function() {
+				// Get the ID of the submodule to render
+				var $this = $(this),
+					args = {
+						parent: container,
+						callback: function() {
+							parent.find('.category').removeClass('loading');
 						}
-					});
-				};
+					},
+					id = $this.attr('id');
+
+				// Display the category we clicked as active
+				parent
+					.find('.category')
+					.removeClass('active')
+					.addClass('loading');
+				$this.toggleClass('active');
+
+				// Empty the main container and then render the submodule content
+				container.empty();
+				monster.pub('voip.' + id + '.render', args);
+			});
+		},
+
+		crossSiteMessageHandler: function(topic) {
+			var crossSiteMessageTopic = topic.replace('.tab', '') + '.render';
 
-			monster.util.getFeatureConfig('smartpbx.other.eventRouting', false)
-				? bindEventNavigation()
-				: bindDefaultNavigation();
+			monster.pub(crossSiteMessageTopic);
```

Comment:
`parent` parameter is missing. 
```suggestion
			monster.pub(crossSiteMessageTopic, { parent: container });
```